pyvista/plotting/_plotting.py

```diff
@@ -5,17 +5,22 @@
 
 import pyvista
 from pyvista.utilities import get_array
+from pyvista.utilities.misc import static_vars
 
 from .tools import opacity_transfer_function
 
 
+@static_vars(_imported=None)
 def _has_matplotlib():
-    try:
-        import matplotlib  # noqa
+    if _has_matplotlib._imported is None:
+        try:
+            import matplotlib  # noqa
+
+            _has_matplotlib._imported = True
+        except ImportError:  # pragma: no cover
+            _has_matplotlib._imported = False
 
-        return True
-    except ImportError:  # pragma: no cover
-        return False
+    return _has_matplotlib._imported
```

Comment:
I'll be honest, this "slapping on attributes at runtime" thing with a decorator is pretty iffy. Looking at the current use cases in this PR, can't we use caching instead?
```py
from functools import cache

@cache
def _has_matplotlib():
    try:
        import matplotlib  # noqa
    except ImportError:  # pragma: no cover
        return False
    return True
```

I haven't tested it but I don't see why this wouldn't work.

Reply:
Implemented this in 730f4a548c74d5aa5778b8d7cdc5a9ae3f1a2a92.

I went for functions rather than a module level constant. A module level constant would require `_has_matplotlib` to be evaluated, which completely defeats our lazy import of `matplotlib`.